src/core/server/server.cc

```diff
@@ -1005,6 +1003,9 @@ grpc_error_handle Server::SetupTransport(
     chand->InitTransport(Ref(), std::move(*channel), cq_idx, transport,
                          channelz_socket_uuid);
   }
+  if (socket_node != nullptr) {
+    channelz_node_->AddChildSocket(socket_node);
```

Comment:
Prior to this PR, we were doing this before calling `chand->InitTransport()`, and now we're doing so afterward.  Are we sure that's safe?  From a quick glance, it's not clear to me whether it's possible for the `ChannelData` dtor (which removes this channelz linkage) to be executed in another thread before we get here, but we need to make sure this happens before that does.

This also makes me realize that it doesn't look like we're removing this channelz linkage anywhere when the connection is closed.  I'm not sure what the right place is do to that, but I think we need to do it somewhere.